### PR TITLE
feat(MiscDrivers): Replace 2.4" TFT fonts with Liberation fonts

### PR DESCRIPTION
**Pull request by vicloginov**
Old fonts for 2.4" TFT are replaced with Liberation fonts in MAX78000/MAX78002 examples.

### CODE REVIEW

**Review by Jake-Carter — COMMENTED** [review]
I think we could consolidate these fonts into `Libraries/MiscDrivers/Display` so we don't have to maintain multiple copies.  I will take that action and move them

**Comment by Jake-Carter** [comment]
@vicloginov @lorne-maxim take a look at the latest changes, especially in `Libraries/MiscDrivers/Display/fonts`

I consolidated the fonts via a `fonts.mk` file.  I included it by default from the AI85 and AI87 BSPs.

```Makefile
# project.mk
FONTS = LiberationSans16x16 LiberationSans28x28
# ...
```

I also added a more generalized `"fonts.h"` in an attempt to abstract away the font-specific pointers better.  Let me know what you think.  Projects using multiple fonts will still need to point to each manually, but that's still possible.

```diff
--- static int font = (int)&Liberation_Sans16x16[0];
+++ static int font = (int)&font_char_table[0];
```

**Comment by Jake-Carter** [comment]
https://github.com/Analog-Devices-MSDK/msdk/actions/runs/6792419214

**Comment by Jake-Carter** [comment]
https://github.com/Analog-Devices-MSDK/msdk/actions/runs/6801968924

File: Examples/MAX78000/CNN/faceid_evkit/project.mk [hunk 1]
```diff
@@ -31,9 +33,10 @@ endif
 # Place build files specific to FTHR_RevA here.
 ifeq "$(BOARD)" "FTHR_RevA"
 # Only Enable if 2.4" TFT is connected to Feather
-#PROJ_CFLAGS+=-DTFT_ENABLE
+PROJ_CFLAGS+=-DTFT_ENABLE
```
Comment (lorne-maxim) [inline Examples/MAX78000/CNN/faceid_evkit/project.mk:36]: Please confirm this change is intentional.

File: Examples/MAX78000/CNN/UNet-highres-demo/project.mk [hunk 2]
```diff
@@ -27,7 +29,7 @@ ifeq "$(BOARD)" "EvKit_V1"
 VPATH += TFT/evkit
 endif
 ifeq "$(BOARD)" "FTHR_RevA"
-VPATH += TFT/fthr
+FONTS = LiberationSans16x16
```
Comment (lorne-maxim) [inline Examples/MAX78000/CNN/UNet-highres-demo/project.mk:32]: Was the VPATH line overwritten intentionally? I see other project.mk files in this PR where VPATH += TFT/fthr was left in.

**Review by lorne-maxim — APPROVED** [review]

File: Examples/MAX78000/CNN/faceid_evkit/project.mk (hunk 1 above)
Comment (Jake-Carter) [reply Examples/MAX78000/CNN/faceid_evkit/project.mk]: Thanks, reverted in https://github.com/Analog-Devices-MSDK/msdk/pull/792/commits/4c7454d5bb69854330aed77f6255a9423ee3addc

File: Examples/MAX78000/CNN/UNet-highres-demo/project.mk (hunk 2 above)
Comment (Jake-Carter) [reply Examples/MAX78000/CNN/UNet-highres-demo/project.mk:32]: Yes, some projects have actual source files place in `TFT/fthr`, whereas others only had the fonts.  I removed the instances where `TFT/fthr` pointed to an empty folder after moving the fonts

File: Libraries/MiscDrivers/Display/fonts/fonts.mk [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+ #
+ ##############################################################################
+# If FONTS_DIR is not specified, this Makefile will locate itself.
+FONTS_DIR ?= $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
```
Comment (Jacob-Scheiffler) [inline Libraries/MiscDrivers/Display/fonts/fonts.mk:19]: Makefile noob here. Can we always guarantee that fonts.mk will be the last Makefile in the list?

Comment (Jake-Carter) [reply Libraries/MiscDrivers/Display/fonts/fonts.mk:19]: Yes, Makefiles are parsed in 2 stages.  In the first stage, it is parsed sequentially through in order.  `include fonts.mk` immediately enter this file and the `FONTS_DIR` variable is created to point to the file's parent directory.

In the second stage, the target rule is run.  This back-propagates through the first run, and recursively resolves any variables that the target depends on.

**Review by Jacob-Scheiffler — APPROVED** [review]

**Comment by vicloginov** [comment]
Compilation of MAX78000 RISC-V examples fails if MAXIM_PATH is unset. It happens due to missing in Makefile.ARM

ifeq "$(MAXIM_PATH)" ""
# MAXIM_PATH is still not defined...
DEPTH := ../../../../
MAXIM_PATH := $(abspath $(DEPTH))
$(warning Warning:  MAXIM_PATH is not set!  Set MAXIM_PATH in your environment or in project.mk to clear this warning.)
$(warning Warning:  Attempting to use $(MAXIM_PATH) calculated from relative path)

**Comment by Jake-Carter** [comment]
Thanks @vicloginov, saw this for some other moved projects.  Fixed most projects as of #799, but will update the RISC-V projects in this PR.  The legacy split makefiles need a manual update

**Comment by vicloginov** [comment]
> Thanks @vicloginov, saw this for some other moved projects. Fixed most projects as of #799, but will update the RISC-V projects in this PR. The legacy split makefiles need a manual update

I still see reference to Liberation_Sans16x16 font in many CNN examples and TFT_Demo like this:

#ifdef BOARD_FTHR_REVA
int image_bitmap_1 = (int)&img_1_rgb565[0];
int image_bitmap_2 = (int)&logo_rgb565[0];
int font_1 = (int)&Liberation_Sans16x16[0];
int font_2 = (int)&Liberation_Sans16x16[0];
#endif

**Comment by Jake-Carter** [comment]
> ifeq "$(MAXIM_PATH)" ""
> DEPTH := ../../../../
> MAXIM_PATH := 
> (
> �
> �
> �
> �
> �
> �
> ℎ
> (DEPTH))

@vicloginov fixed in https://github.com/Analog-Devices-MSDK/msdk/pull/792/commits/69990eea60812ca7692534d90825a87b4043322c

**Comment by Jake-Carter** [comment]
> > Thanks @vicloginov, saw this for some other moved projects. Fixed most projects as of #799, but will update the RISC-V projects in this PR. The legacy split makefiles need a manual update
> 
> I still see reference to Liberation_Sans16x16 font in many CNN examples and TFT_Demo like this:
> 
> #ifdef **BOARD_FTHR_REVA** int image_bitmap_1 = (int)&img_1_rgb565[0]; int image_bitmap_2 = (int)&logo_rgb565[0]; **int font_1 = (int)&Liberation_Sans16x16[0]; int font_2 = (int)&Liberation_Sans16x16[0];** #endif
> 
> It causes compilation error for BOARD=FTHR_RevA

@vicloginov which examples?  `TFT_Demo` builds for me successfully

```serial
~/repos/msdk/Examples/MAX78000/TFT_Demo (feat/fonts*) » make BOARD=FTHR_RevA                          jhcarter@jhcarter-X670E
Loaded project.mk
****************************************************************************
* Analog Devices MSDK
* - User Guide: https://analog-devices-msdk.github.io/msdk/USERGUIDE/
* - Get Support: https://www.analog.com/support/technical-support.html
* - Report Issues: https://github.com/Analog-Devices-MSDK/msdk/issues
* - Contributing: https://analog-devices-msdk.github.io/msdk/CONTRIBUTING/
****************************************************************************
- MKDIR /home/jhcarter/repos/msdk/Examples/MAX78000/TFT_Demo/build
- CC resources/tft_fthr/img_1_rgb565.c
- CC main.c
- CC src/state.c
- CC src/state_home.c
- CC src/state_info.c
- CC src/state_keypad.c
- CC src/utils.c
- CC /home/jhcarter/repos/msdk/Libraries/Boards/MAX78000/FTHR_RevA/Source/board.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/stdio.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/LED/led.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/PushButton/pb.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Display/tft_ili9341.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Camera/camera.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Camera/ov7692.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Camera/sccb.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/PMIC/max20303.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/CODEC/max9867.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Display/fonts/LiberationSans12x12.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Display/fonts/LiberationSans16x16.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Display/fonts/LiberationSans19x19.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Display/fonts/LiberationSans24x24.c
- CC /home/jhcarter/repos/msdk/Libraries/MiscDrivers/Display/fonts/LiberationSans28x28.c
- AS /home/jhcarter/repos/msdk/Libraries/CMSIS/Device/Maxim/MAX78000/Source/GCC/startup_max78000.S
- CC /home/jhcarter/repos/msdk/Libraries/CMSIS/Device/Maxim/MAX78000/Source/heap.c
- CC /home/jhcarter/repos/msdk/Libraries/CMSIS/Device/Maxim/MAX78000/Source/system_max78000.c
- LD /home/jhcarter/repos/msdk/Examples/MAX78000/TFT_Demo/build/max78000.elf
arm-none-eabi-size --format=berkeley /home/jhcarter/repos/msdk/Examples/MAX78000/TFT_Demo/build/max78000.elf
   text	   data	    bss	    dec	    hex	filename
 226980	   1736	   2388	 231104	  386c0	/home/jhcarter/repos/msdk/Examples/MAX78000/TFT_Demo/build/max78000.elf
```

**Comment by Jake-Carter** [comment]
Manually running auto builder, not sure why it's not triggering for this PR.

https://github.com/Analog-Devices-MSDK/msdk/actions/runs/6898675901

**Pull request merged**